lib/name.ml

```diff
@@ -93,15 +93,16 @@ let parse names base buf = (* what. a. mess. *)
           aux (o :: offsets) (n :: name) (base+offset) (Cstruct.shift buf offset)
 
       | (P (p, _), offset) ->
-          let ns = (Hashtbl.find_all names p
-                       |> List.filter (function L _ -> true | _ -> false))
-          in
+          let all = Hashtbl.find_all names p in
+          if ((List.length all) == 0) then
```

Comment:
I suspect you do since this is in a `match` (and there are more cases of the outer match following)